src/pal/src/misc/miscpalapi.cpp

```diff
@@ -298,3 +301,153 @@ PAL_Random(
     PERF_EXIT(PAL_Random);
     return bRet;
 }
+
+
+// Support for EnsureOpenSslInitialized
+
+static const char * const libcryptoName = "libcrypto" PAL_SHLIB_SUFFIX;
```

Comment:
PAL_SHLIB_SUFFIX is .so on Linux, and you will not get libcrypto.so unless you have the -dev package installed, which is probably not a dep we want to take.

```
vagrant@vagrant-ubuntu-trusty-64:~$ dpkg -L libssl0.9.8 | grep libcrypto
/lib/x86_64-linux-gnu/libcrypto.so.0.9.8
/usr/lib/x86_64-linux-gnu/libcrypto.so.0.9.8
vagrant@vagrant-ubuntu-trusty-64:~$ dpkg -L libssl-dev | grep libcrypto
/usr/lib/x86_64-linux-gnu/libcrypto.a
/usr/lib/x86_64-linux-gnu/pkgconfig/libcrypto.pc
/usr/lib/x86_64-linux-gnu/libcrypto.so
```